plugin/trino-memsql/src/test/java/io/trino/plugin/memsql/TestMemSqlTypeMapping.java

```diff
@@ -567,19 +522,32 @@ private DataTypeTest dateTestCases(DataType<LocalDate> dateDataType, ZoneId jvmZ
         LocalDate dateOfLocalTimeChangeBackwardAtMidnightInSomeZone = LocalDate.of(1983, 10, 1);
         verify(someZone.getRules().getValidOffsets(dateOfLocalTimeChangeBackwardAtMidnightInSomeZone.atStartOfDay().minusMinutes(1)).size() == 2);
 
-        return DataTypeTest.create()
-                .addRoundTrip(dateDataType, LocalDate.of(1, 1, 1))
-                .addRoundTrip(dateDataType, LocalDate.of(1582, 10, 4)) // before julian->gregorian switch
-                .addRoundTrip(dateDataType, LocalDate.of(1582, 10, 5)) // begin julian->gregorian switch
-                .addRoundTrip(dateDataType, LocalDate.of(1582, 10, 14)) // end julian->gregorian switch
-                .addRoundTrip(dateDataType, LocalDate.of(1952, 4, 3)) // before epoch
-                .addRoundTrip(dateDataType, LocalDate.of(1970, 1, 1))
-                .addRoundTrip(dateDataType, LocalDate.of(1970, 2, 3))
-                .addRoundTrip(dateDataType, LocalDate.of(2017, 7, 1)) // summer on northern hemisphere (possible DST)
-                .addRoundTrip(dateDataType, LocalDate.of(2017, 1, 1)) // winter on northern hemisphere (possible DST on southern hemisphere)
-                .addRoundTrip(dateDataType, dateOfLocalTimeChangeForwardAtMidnightInJvmZone)
-                .addRoundTrip(dateDataType, dateOfLocalTimeChangeForwardAtMidnightInSomeZone)
-                .addRoundTrip(dateDataType, dateOfLocalTimeChangeBackwardAtMidnightInSomeZone);
+        for (String timeZoneId : ImmutableList.of(UTC_KEY.getId(), jvmZone.getId(), someZone.getId())) {
+            Session session = Session.builder(getSession())
+                    .setTimeZoneKey(TimeZoneKey.getTimeZoneKey(timeZoneId))
+                    .build();
+
+            SqlDataTypeTest.create()
+                    .addRoundTrip("date", "CAST('0001-01-01' AS date)", DATE, "DATE '0001-01-01'")
```

Comment:
Is this minimum supported date by MemSQL? If so add a comment otherwise let's test negative dates too.

Reply:
We should also test max date similarly.

Reply:
Let me handle negative dates in #10320